### DISCUSSION TITLE
DATABASE npl_class: Parsing and Consolidation ?

### DISCUSSION THREAD

**Discussion by cverluise** [cverluise/PatCit#27] [Projects & Ideas]
Hello,

Since v0.2-npl, we have introduced a DATABASE npl_class

### Question
How should this class of citations be structured and consolidated?

### Food for thoughts
Below is a sample of 100 citations for this class.

<details>

```sql
WITH
  tmp AS (
  SELECT
    npl_publn_id
  FROM
    `npl-parsing.patcit.v02_npl`
  WHERE
    npl_class="DATABASE")
SELECT
  npl_biblio
FROM
  `usptobias.patstat.tls214` AS tls214,
  tmp
WHERE
  tmp.npl_publn_id = tls214.npl_publn_id
LIMIT 100  
```

Row | npl_biblio |  
-- | -- | --
1 | GENESEQ Database, Search Date Sep. 8, 2003 Accession No. P AAE04230 (Ruben SM et al.), Pub Date May 25, 2001. |  
2 | Lerner et al., GenBank Accession No. AAB42052 (U39534.1), submitted Oct. 27, 1995. |  
3 | Database Registry Accession No. RN 922391-60-4; Entered STN: Feb. 22, 2007. |  
4 | Pritzlaff et al., 'Streptococcus agalactiae cyl gene cluster, partial sequence,' GenBank Accession No. AF157015, Feb. 8, 2001. |  
5 | GenBank Accession No. U91324 for, Human chromosome 2 BAC clone CIT987SK 100B4 complete sequence Mar., 1997. |  
6 | NCBI Sequence NM-003012, 'Homo sapiens secreted frizzled-related protein 1 (SFRP1), mRNA,' pp. 1-6, http://www.ncbi.nih.gov, printed May 5, 2006. |  
7 | GenBank: AY383628.1. Synthetic construct HPV18 major capsid protein L1 (L1) gene, complete cds. Published on Oct. 11, 2003. |  
8 | Chong, S.S., et al. (GI 450532), GenBank Sequence Database (Accession X71355), National Center for Biotechnology Information, National Library of Medicine, Bethesda, Maryland, 20894. No date provided. |  
9 | NCBI annotation project, GenBank Accession Number, XP-002252, Oct. 2001. |  
10 | DATABASE UNIPROTKB [online] 20 December 2005 (2005-12-20), retrieved from http:l/www.uniprot.org/uniprot/POAFQO Database accession no. POAFQO |  
11 | UniProt Accession No. D1MAM0, Full=Tyrosine-protein kinase receptor; EC=2 . 7 . 10 . 1, 2010, XP-002656286, two pages. |  
12 | Database accession No. RN 1144428-37-4; XP-002745331; (May 8, 2009) Supplier: Interbioscreen Ltd.; 1 page. |  
13 | Database Swissprot-42 Accession No. Q57124, Nov. 10, 1997, 1 page. |  
14 | Database STN Abstract: Accession No. 1984:603807 CAPLUS, Document No. 101-203807, 'Metabolism and disposition of 3-amino-1,5-dihydro-5-methyl-1-beta-D-ribofuranosyl-1,4,5,6,8-pentaazaacenaphthylene in the rat', Basseches et al., Cancer Research 1984, 44(9), 3672-8. |  
15 | GenBank Accession No. CJ669561 'CJ669561 Y.Ogihara unpublished cDNA library Wh-KMV Triticum aestivum cDNA clone whkv19d01 5-, mRNA. sequence' dated Feb. 23, 2013. |  
16 | Genebank Accession No. CY041637. |  
17 | GENESEQ Database, Search Date Sep. 8, 2003 Accession No. P AAB31202 (Ashkenazi AJ et al.), Pub Date Dec. 21, 2000. |  
18 | GenBank Accession No. XP-001273036.1, published Feb. 21, 2008. |  
19 | Database Uniprot Database accession No. B9SXX3. |  
20 | DATABASE PUBCHEM. [o] 13 February 2015 (2015-02-13), Database accession no. 15257684. |  
21 | Geneseq [Online] Oct. 23, 1998, 'Streptococcus pneumoniae genome fragment SEQ ID No. 214.' retrieved from EBI accession No. GSN: AAV52347, Database accession No. AAV52347 (the whole document). |  
22 | Database Registry [Online] Chemical Abstracts Service, Columbus, Ohio, US; Jun. 10, 2008, 'Not yet assigned', Database accession No. 1027012-36-7 the whole document. |  
23 | DATABASE UniProt EMBL; 1 November 1997 (1997-11-01), retrieved from UNIPROT accession no. Q24733 |  
24 | NCBI Entrez, Genbank Report, Accession No. AA154256, Marra, M. et al. (Feb. 1997). |  
25 | Marra et al., 1997, GenBank Accession No. AA170532. |  
26 | GenBank Accession No. AA281296. |  
27 | NCBI GenBank Accession No. EU216059, Homo sapiens BCR/ABL fusion protein isoform X2 (BCR/ABL fusion) mRNA, complete cds; http://www.ncbi.nlm.nih.gov/nuccore/EU216059.1?report=gbwithparts&log$=seqview, printed Aug. 8, 2012 2 pages. |  
28 | Genbank Entry, Accession No. AA010038 (1996). |  
29 | Geneseq Database Accession No. Y00771 (Jan. 21, 1999). |  
30 | NCBI Reference sequence: XP-001164530, Sep. 15, 2006. |  
31 | Genbank Accession No. AI492143 (Mar. 30, 1999). |  
32 | Database EMBL, entry HSGS04157, Accession No. D25789, Nov. 27, 1995. |  
33 | 'GENBANK', Database accession no. AAF172491 |  
34 | GenBank Accession No. AB333793.1 (2008). |  
35 | GenBank Accession No. AA178778 (1997). |  
36 | Genbank Submission; NCBI, Accession No. NP-049542; Xiao et al.; Mar. 11, 2010. |  
37 | GenPept, Accession No. CAA72296, Mouillet J.F., Tenebrio molitor mRNA for ecdysone receptor. |  
38 | GenBank Accession No. U65416 (gi:1815636), 'Human MHC class I molecule (MICB) gene, complete cds,' Feb. 5, 1997. |  
39 | NCBI Database Report for Accession No. BAD59440, Direct Submission on Sep. 26, 2003. |  
40 | Wilson (2007, GenBank Accession No. AC 190815). |  
41 | GenBank Accession No. Q9WWD3, created Nov. 1, 1999. |  
42 | Fleischmann, R.D. et al., (GI 1574563), GenBank Sequence Database (Accession U32844; L42023), National Center for Biotechnology Information, National Library of Medicine, Bethesda, Maryland, 20894. |  
43 | ‘Duck Tales,’(1987)[TV Series 1987-1990], Internet Movie Database (IMDB) [Retrieved on Apr. 7, 2007]. |  
44 | Hillier et al., Genbank Accession No. AA149044 (May 19, 1997). |  
45 | Marra et al., 1997, GenBank Accession No. AA423158. |  
46 | NCBI NP-075598.2 Basic Fibroblast growth factor receptor 1 isoform 1 precursor [Homo sapiens] [online] Dec. 13, 2009. Available on the internet: http://www.ncbi.nlm.nih.gov/protein/NP-075598.2. |  
47 | NCBI Locus Report for TP43 (www.ncbi.nlm.gov/gene?term=7157) Last visited Sep. 29, 2014. |  
48 | GenBank Submission; NIH/NCBI, Accession No. NM-014707, Muralidhar et al.; Mar. 11, 2011. |  
49 | GenBank Accession AC007869 |  
50 | Database EMBL Accession No. P02643 Jul. 21, 1986 XP-002275449. |  
51 | 'Topic Real-Time', HP-UX-Documentation Disc 50726-10186 (from Software Patent Institute Database of Software Technologies), Feb. 1, 1995. |  
52 | GenBank accession number WP_003406912; 6/22/2013 |  
53 | GenBank Accession No. AEK08729 dated Feb. 23, 2005, 2 pages. |  
54 | DATABASE REGISTRY, vol. RN 202207-27-0, JPN6018008407, 5 March 1998 (1998-03-05) |  
55 | GenBank Accession No. AA532364 for nj12a08.s1 NCI<SUB>-</SUB>CGAP<SUB>-</SUB>Pr22 Homo sapiens cDNA clone IMAGE:986102 3', mRNA sequence. |  
56 | EMBL Sequence Data Library Accession No: O48889, Jun. 01, 1998, Bolchi, A. et al. |  
57 | Database Registry, XP-002336813, May 19, 1989, RN-120685-11-2, 2 pages. |  
58 | Albrandt et al., 'Human Uncoupling Protein 4,' Database: A_Geneseq_36, Accession No. Y78511, 2000. |  
59 | 《GenBank》 20081204 Kikuchi,S.等 AK062614 序列表 1，2 , 2 |  
60 | Database Caplus on STN: Accession No. 1976-519336 (1976). |  
61 | GenBank GI No. 116875767; Accession No. NP 003248.3, dated Oct. 31, 2006, 7 pages. |  
62 | Yin, D.X. et al., (Direct Submission), GenBank Sequence Database (Accession U66879), National Center for Biotechnology Information, National Library of Medicine, Bethesda, Maryland, 20894 (GI 1683636; GI 1683637) Nov. 23, 1996. |  
63 | Hillier, L. et al. Wash U-Merck EST Project Genbank Accession No. 1264408 May 26, 1995. |  
64 | GENBANK Accession No. Z32768 1994 'Human coronavirus OC43 gene for surface protein', 3 pages. |  
65 | Pring-Akerblon, Hexon Protein-Human Adenovirus 4, Accession No. S57637, submitted to the EMBL Data Library, (Feb. 1995). |  
66 | Database accession No. 147:166330 abstract, Database CAPLUS, Chemical Abstracts Service, Columbus Ohio, US, Retrieved from STN accession No. 2007:711051, 2007, 1 page. |  
67 | GenBank Accession No. AA746653 for Homo Sapiens cDNA clone 1257327; Jan. 22, 1990. |  
68 | DATABASE EMBL DATABASE 'Extract V00320' |  
69 | GenBank Accession No. AB020693 dated Jan. 10, 2004. |  
70 | GenBank Submission; NIH/NCBI, Accession No. EAS23724; Mar. 22, 2006. |  
71 | Ink A Dink A Do, 4 Inks in One, Stamp Pad (undated). |  
72 | GenBank Accession No: AA242576, Mar. 7, 1997. |  
73 | Haecker, Genbank Accession M96441 Medline 92409595, May 5, 1993. |  
74 | Database Biosis, Biosciences Information Service, Philadelphis, PA, k. Fujita et al., 'Effects of flunarizine on primary open angle and low tension glaucomas'. |  
75 | Genbank No. AAG10021 dated Sep. 3, 2000. |  
76 | Sasaki T., EMBL Database, Aug. 6, 1997, Rice CDNA from callus, XP00212812. |  
77 | Genbank Accession No. W97816 (Soares mouse embryo NbME13.5 14.5 cDNA) Jul. 16, 1996. |  
78 | NCBI Entrez Accession No.: gil14149916, submitted May 2000. |  
79 | Database GenBank (EST); Accession No: A1987883; Marra et al. mouse clone; Sep. 2, 1999; Having at least 18 and 50 contiguous nucleotides of SEQ ID NO: 5. |  
80 | NCBI, (mRNA Sequence) Accession No. AI925946, 'wh12f04.x1 NCI-CGAP-Kid11 Homo sapiens cDNA clone Image: 2380543 3- similar to TR:O00581 O00581 Hypothetical 20.5 KD Protein.', [retrieved on Apr. 19, 2011] Retreived from the Internet <URL:http://www.ncbi.nlm.gov/nucest/5661910?report=genbank. |  
81 | EMBL Database Report for Accession No. 24995, dated Nov. 9, 1992. |  
82 | Database EBI Accession No. UniProt: A1JRX1, Feb. 6, 2007, Isocitrate lyase, XP-002576046. |  
83 | GenBank Accession No. AK001872.1, 'Homo sapiens cDNA FLJ11010 fis, clone PLACE1003145', pp. 1-2, (submitted Feb. 16, 2000). |  
84 | GenBank Accession No. AK055917, published Jan. 19, 2008. |  
85 | GenBank accession 000375 Jun. 1997. |  
86 | Net core GenBank accession No. AF069670. |  
87 | NCBI (2005) Accession NP-566186. |  
88 | GenBank Accession No. AE000554. Aug. 6, 1997. |  
89 | Database Caplus (Online), Chemical Abstracts Service, XP000002657265, Database accession No. 2005:1235680 abstract & CN 1 596 742 A, (2005). |  
90 | Rouse et al. GenBank Accession No. A54805, Oct. 28, 1994. |  
91 | GenBank Accession No. AF490589, Arabidopsis thaliana Na<SUP>+</SUP>/H<SUP>+ </SUP>exchanger 5 (NHX5) mRNA, complete cds, 2002. |  
92 | Vogel, P. et al., (GI 1834507), GenBank Sequence Database (Accession Z50781), National Center for Biotechnology Information, National Library of Medicine, Bethesda, Maryland, 20894. |  
93 | GenBank Accession No. Q9Y5S2. |  
94 | NCBI Accession No. P08176(DerP1) 9 pages. |  
95 | Gray, et al., Maturase [Neosartorya fischeri] GenBank Accession No. AAX39426, May 19, 2005. |  
96 | GENBANK nucleotide sequence AB008394 (Mar. 14, 1009). |  
97 | EMBL Database, Accession No. U80669, Prescott, J.S. et al. 'Human androgen regulated homeobox protein (NKX3.1) mRNA, complete cds' (Dec. 1996). |  
98 | DATABASE GENBANK 5 March 1996 (1996-03-05), CHEN XY. ET. AL. |  
99 | Genbank Accession No. AI867518 (Dec. 17, 1999). |  
100 | GenBank Accession No. CAB03914 for C24H11.8, Nov. 23, 1998.

</details>

### Open questions

- Need for further sub-classification? If yes, which sub-classes?
- Fields to be parsed?
- Corpus/ reference database for consolidation?

I leave these questions open to discussion.

Cheers

**Comment by gderasse** [comment]
This is tricky. We could try searching for the database on https://datasetsearch.research.google.com/ but I am not convinced it is a fruitful direction. 
1/ Many entries are related to GenBank, so we could probably seek to consolidate NPL for a few selected databases. 
2/ Otherwise, we could ask the EPO for a list of the databases they have access to. After all, examiners must have access to these databases for performing examination.

**Comment by cverluise** [comment]
Some thoughts on the parsing approach:

- **database_name** (e.g. GENBANK, EMBL, NCBI, etc)
- **accession_number** (e.g. Y78511, 1976-519336, etc) -> key for consolidation
- **accession_year** (e.g. 2006)
- **bibliographical_reference** (e.g Prescott, J.S. et al. 'Human androgen regulated homeobox protein (NKX3.1) mRNA, complete cds' (Dec. 1996), row 97)
- **locus** (e.g. C24H11.8, NKX3.1) -> key for consolidation? enrichment?
